jsonLoader.go

```diff
@@ -100,25 +104,32 @@ func (l *jsonReferenceLoader) JsonReference() (gojsonreference.JsonReference, er
 	return gojsonreference.NewJsonReference(l.JsonSource().(string))
 }
 
+func (l *jsonReferenceLoader) AddSchema(uri string, b []byte) {
```

Comment:
A `JsonLoader` is an abstract way of loading JSON for both schema and data JSON, so it breaks the abstraction to suddenly have one function that is specific to one of the two cases. If you'd want to do it at the `Schema` level you're already too late, because `NewSchema` already parses the entire schema and all the derefencing already happened. Sadly there is really no way to do it different way without significantly altering the API in a backwards incompatible way, so this is probably the right place to add this functionality.

Anyway, since `JSONLoader`s are used everywhere to input JSON in various ways, you'd expect that if you add a schema manually a `JSONLoader` would be required, not `[]byte`. It's also far more flexible.